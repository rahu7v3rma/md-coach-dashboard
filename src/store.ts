import { configureStore } from '@reduxjs/toolkit';

import clientReducer from './reducers/client';
import groupReducer from './reducers/group';
import notificationReducer from './reducers/notification';
import userReducer from './reducers/user';

// define store options as a variable so we can export it for tests to use
const storeOptions: any = {
    reducer: {
        user: userReducer,
        client: clientReducer,
        notification: notificationReducer,
        group: groupReducer
    }
};

const store = configureStore(storeOptions);

export default store;

// infer the `RootState` type from the store
export type RootState = ReturnType<typeof store.getState>;

// infer the `AppDispatch` type from the store so thunk actions can return promises
export type AppDispatch = typeof store.dispatch;

export const test = {
    storeOptions
};
