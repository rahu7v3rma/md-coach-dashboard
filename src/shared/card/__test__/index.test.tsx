import React from 'react';
import { Provider } from 'react-redux';
import * as router from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { act, create } from 'react-test-renderer';

import Card from '..';
import store from '../../../store';

const navigate = jest.fn();

describe('Card', () => {
    beforeEach(() => {
        jest.spyOn(router, 'useNavigate').mockImplementation(() => navigate);
        jest.clearAllMocks();
    });

    it('Card snapshot', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Card />
                </BrowserRouter>
            </Provider>
        );
        expect(tree).toMatchSnapshot();
    });

    it('Card renders without errors', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Card />
                </BrowserRouter>
            </Provider>
        );
        expect(tree.root).toBeTruthy();
    });

    it('should apply custom styles when props are provided', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Card width="335px" height="100px" />
                </BrowserRouter>
            </Provider>
        );
        let cardContainer = tree.root.findByProps({
            width: '335px'
        }).props;
        expect(cardContainer).toBeTruthy();
        expect(cardContainer.width).toBe('335px');
        expect(cardContainer.height).toBe('100px');
    });

    it('should handle click events', async () => {
        const onClick = jest.fn();
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Card width="335px" onClick={onClick} />
                </BrowserRouter>
            </Provider>
        );
        let cardContainer = tree.root.findByProps({
            width: '335px'
        }).props;
        expect(cardContainer).toBeTruthy();
        await act(() => cardContainer.onClick());
        expect(onClick).toHaveBeenCalledTimes(1);
    });

    it('should render children', async () => {
        const tree = create(
            <Provider store={store}>
                <BrowserRouter>
                    <Card>
                        <div id="child" />
                    </Card>
                </BrowserRouter>
            </Provider>
        );
        let child = tree.root.findByProps({
            id: 'child'
        }).props;
        expect(child).toBeTruthy();
    });
});
