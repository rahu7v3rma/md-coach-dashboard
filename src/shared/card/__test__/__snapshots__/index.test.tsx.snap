// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card snapshot 1`] = `
<div
  className="sc-bgqQcB jOQRxz"
/>
`;
